### PR TITLE
Bearer tokens

### PR DESCRIPTION
**Pull request by betaredex**
Add support for bearer token authentication
Use default authentication settings for both pman and pfioh

### CODE REVIEW

**Comment by rudolphpienaar** [comment]
Is this OK to merge? I'm catching up on the current state of PRs etc...

**Comment by betaredex** [comment]
No, it's not fully tested yet

**Comment by danmcp** [comment]
@betaredex Please squash the commits

File: pfcon/pfcon.py [hunk 1]
```diff
@@ -116,16 +120,30 @@
...
+                "baseURLpath":                  "api/v1/cmd/", 
                 "status":                       "undefined"
             },
             "data": {
                 "addr":                         "pfioh-myproject.127.0.0.1.nip.io",
                 "baseURLpath":                  "api/v1/cmd/",
-                "allowUnverifiedCertificates":  False,  
                 "status":                       "undefined"
             }
+        },
```
Comment (danmcp) [inline pfcon/pfcon.py]: Does this still make sense?  Shouldn't all the examples pass default token?

File: pfcon/pfcon.py [hunk 2]
```diff
@@ -434,29 +448,50 @@ def dataRequest_process(self, *args, **kwargs):
...
-            msg                         = json.dumps(d_request),
-            verb                        = 'POST',
-            http                        = '%s/%s' % (str_dataServiceAddr, str_dataServiceURL),
-            b_quiet                     = False,
-            b_raw                       = True,
-            b_httpResponseBodyParse     = True,
-            jsonwrapper                 = ''
-        )
-
+        b_enableTokenAuth       = Gd_tree.cat('/service/%s/data/enableTokenAuth'% str_remoteService)
```
Comment (danmcp) [inline pfcon/pfcon.py]: Do we still need the enable auth option at all now?  Can we not just say it's always enabled and the default token is passed unless you change it?

File: pfcon/pfcon.py [hunk 3]
```diff
@@ -24,9 +24,11 @@
 import  os
 import  multiprocessing
 import  pfurl
+import  configparser
 
 import  pfmisc
 
+# debugging utilities
 import  pudb
 import  swiftclient
```
Comment (danmcp) [inline pfcon/pfcon.py]: the swiftclient is for debugging?

File: pfcon/pfcon.py [hunk 4]
```diff
@@ -1201,7 +1260,7 @@ def compute_handler():
             if str_outDirPath is not None:
                 # This value will not be none in case of non-swift option.
                 str_outDirParent, str_outDirOnly = os.path.split(str_outDirPath)
-            # pudb.set_trace()
+            #
```
Comment (danmcp) [inline pfcon/pfcon.py]: Maybe remove these empty comments

File: pfcon/pfcon.py [hunk 5]
```diff
@@ -1246,7 +1305,7 @@ def jobStatusFiles_create():
             # information about the swift operation. We push them to object storage
             # after creating them in the pfcon FS). This means that the files may 
             # not be registered with CUBE since when CUBE might pull from swift storage
-            # these last two files might not be consistently reported to the pulling
+            # these last two files might not be conpushData_sistently reported to the pulling
```
Comment (danmcp) [inline pfcon/pfcon.py]: Guessing this was in error

File: pfcon/pfcon.py [hunk 6]
```diff
@@ -75,28 +76,29 @@
...
     'service':  {
         'host': {
             'data': {
-                'addr':         '%PFIOH_IP:5055',
-                'baseURLpath':  'api/v1/cmd/',
-                'status':       'undefined'
+                'addr':            '%PFIOH_IP:5055',
+                'baseURLpath':     'api/v1/cmd/',
+                'status':          'undefined',
+                "authToken":      "password"
```
Comment (danmcp) [inline pfcon/pfcon.py]: Not aligned and use '' consistently

**Comment by danmcp** [comment]
@betaredex Looks good.  Please fix the conflict.

**Comment by danmcp** [comment]
@rudolphpienaar Please review

**Comment by rudolphpienaar** [comment]
LGTM. I guess squash the commits?

**Comment by betaredex** [comment]
The other commit is Emilio's, not mine.

**Pull request merged**